### DISCUSSION TITLE
[Pre-release] v8.1.240606-rc

### DISCUSSION THREAD

**Discussion by michael-hawker** [CommunityToolkit/Windows#429] [Announcements]
### Important Notes

* There should also be a better error message now if the wrong TFM is used by a consuming project:

![image](https://github.com/CommunityToolkit/Windows/assets/24302614/e512dfbc-ec7a-447f-9594-613e31cd7c29)

* We may have noticed issues with the transitive dependencies for the `Media` packages (or possibly related to `Animations` package), if you notice issues where you need to install additional packages manually, [please comment on this issue](https://github.com/CommunityToolkit/Windows/issues/169#issuecomment-2153140866). 

### Highlights
* Enable AoT for net8 in Release by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/417
* Update Uno version, Uno TFMs to .NET 8 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/422
* fix: Uno Platform - Avoid square border around `RangeSelector` `Thumbs` by @MartinZikmund in https://github.com/CommunityToolkit/Windows/pull/400
* Fix DockPanel measuring bug by @Poker-sang in https://github.com/CommunityToolkit/Windows/pull/416

### Documentation
* Update DockPanel.md by @mael8 in https://github.com/CommunityToolkit/Windows/pull/419
* Documentation fixup for v8.1  by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/398
* Fixed remaining Microsoft Learn doc warnings by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/410

### Infrastructure
* Updated actions to latest version for node 20 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/387
* Fixed duplicate artifact names across matrices in CI by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/411
* Fixed duplicated artifacts in CI when uploading package list by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/421

## New Contributors
* @MartinZikmund made their first contribution in https://github.com/CommunityToolkit/Windows/pull/400
* @mael8 made their first contribution in https://github.com/CommunityToolkit/Windows/pull/419

**Full Changelog**: https://github.com/CommunityToolkit/Windows/compare/v8.1.240328-rc...v8.1.240606-rc

For changelog of the last 8.1-rc compared to the last stable release: https://github.com/CommunityToolkit/Windows/releases/tag/v8.1.240328-rc

<hr /><em>This discussion was created from the release <a href='https://github.com/CommunityToolkit/Windows/releases/tag/v8.1.240606-rc'>v8.1.240606-rc</a>.</em>

**Comment by Joost-Jens-Luminis** [comment]
For some reason it can't find the Behaviors or Converters namespace now after the update. Has that been moved or do I have to update something else as wel?

I do have the package installed:
![image](https://github.com/user-attachments/assets/63fc062a-9438-47dc-a591-c73f97dd5334)

**Reply by Joost-Jens-Luminis** [reply]
I figured it out, we were still on version 19041 for your minimum and target versions, but this has to become 22621.

**Reply by michael-hawker** [reply]
Glad you figured it out, did you see the improved error message called out above? (Did that help?)

![](https://private-user-images.githubusercontent.com/24302614/337368482-e512dfbc-ec7a-447f-9594-613e31cd7c29.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NTU5NDEsIm5iZiI6MTcyMzc1NTY0MSwicGF0aCI6Ii8yNDMwMjYxNC8zMzczNjg0ODItZTUxMmRmYmMtZWM3YS00NDdmLTk1OTQtNjEzZTMxY2Q3YzI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDIxMDA0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0OTJhNWZiNWM4ZGFkYzU4ODBmNDA2MWJlZDNhZTNjMGE3NWRmYTdjY2MyZWYyNzMwNDRiYTEyNjUyY2Y1ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WI29f_jEPwJz0Su-Tts1dLbW696vHf4b6dUU15NeN1Y)

This is one of the main upgrade issues we've noticed folks having, so we've been trying to call it out in each of our release notes.

**Reply by Joost-Jens-Luminis** [reply]
I did not see the improved error message. I double checked for you and it doesn't appear in the build output.
It might be better to set this new requirement explicitly and clearly at the beginning of the release notes. Right now it seems to be required only for the media package, but we are not using that so that causes confusion.

Also, you have to go to the What's Changed page to see this more clearly. The release notes are even less clear about this.

**Comment by Joost-Jens-Luminis** [comment]
As a result of changing the TFM to version 22621, our APP is longer installing on older windows versions (19041). We have set the TargetFrameworkVersion to 22621 but have left the SupportedOSVersion on 19041.
![image](https://github.com/user-attachments/assets/78755936-9cfa-4636-90e6-b96d0ff0cb41)

I saw in the release notes that this was as a result of upgrading to WindowsAppSDK 1.4/1.5, however we were already using the latest WindowsAppSDK version and were targeting TFM 19041 for the last year.
 
At the moment this is no longer working with this RC and as such we can no longer support any windows 10 version older then the latest one. This is causing a problem for us as some of our clients refuse or are unable to upgrade.
According to the release notes it should still work with older Windows versions without issues.
![image](https://github.com/user-attachments/assets/2fed6378-547b-4653-aabc-26fad741ae1c)

Is this behaviour intended? Can you have the SupportedOSVersion set to 19041 while the target is 22621?

**Reply by Arlodotexe** [reply]
19041 is still our minimum, set by `TargetPlatformMinVersion` [here](https://github.com/CommunityToolkit/Tooling-Windows-Submodule/blob/fda081e11081ec7b0c034cdacbb817d906128b24/MultiTarget/WinUI.Extra.props#L12). 

@michael-hawker Looking at the docs for [`SupportedOSPlatformVersion`](https://learn.microsoft.com/en-us/dotnet/standard/frameworks#support-older-os-versions), it seems like `TargetPlatformMinVersion` could be specific to uap while `net*` TFMs might use `SupportedOSPlatformVersion` instead? That seem right?

**Reply by michael-hawker** [reply]
Yeah, maybe we need both if we don't already have them? There's a warning if they're mismatched: https://learn.microsoft.com/dotnet/core/tools/sdk-errors/netsdk1135

@Sergio0694 do you have some insights here for us?

**Reply by Sergio0694** [reply]
`TargetPlatformMinVersion` doesn't do anything on modern .NET, it has been renamed to `SupportedOSPlatformVersion`, which generates the new attribute and enables the analyzers. Pretty much on modern .NET you should just drop TPMV entirely, and set this new property instead. See change [here](https://github.com/dotnet/designs/pull/157), and related design doc :)

**Reply by Joost-Jens-Luminis** [reply]
hmm, that is interesting. When the only thing we have changed is the TargetFramework and upgrade the CommunityToolkit packages, why does the app no longer install on 19041?

**Reply by Joost-Jens-Luminis** [reply]
I have just reverted the CommunityToolkit packages to the release version, but left the rest of my changes and I still can't install it. So at least that doesn't seem to be the fault of the Toolkit directly, but it does mean it can't be run on older OS versions anymore.

**Reply by michael-hawker** [reply]
@Joost-Jens-Luminis we weren't setting this property in the Toolkit, so I think that's why you were having issues. We're testing that out now. Thanks for letting us know! See https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/206

**Reply by Sergio0694** [reply]
@Joost-Jens-Luminis you can't _run_, or you can't _install_? Those are two very different things. Assuming you can't install, and that your app is packaged, are you sure you didn't accidentally change the target/minimum SDKs in your app manifest? Because that's what would affect not being able to install a given package on an older version of Windows.

**Reply by Arlodotexe** [reply]
>  you can't run, or you can't install?

To repro the issue with the RC packages, this was my next question. Are you able to deploy on a machine running `10.0.19041.0` or greater, but not lower like `10.0.17763.0`? Or are you running into errors while installing the app package?

**Reply by Arlodotexe** [reply]
I've spun up a few VMs for Windows 10 Version 1809 Build 17763 and Windows 10 Version 21H1 Build 19043 to test the RC packages, I'm not seeing any issues building and deploying in a blank WinUI 3 app.

**Reply by Joost-Jens-Luminis** [reply]
I can not install. I haven't tested running yet. Even without the package update I can't install. This means it's not directly an issue with CommunityToolkit. But it does mean the requirement to increase the TFM prevents distribution on lower windows versions via a standard installer or possibly the windows store. 

For clarity, we are creating the installer bundle file as follows via Azure Devops Pipelines:
  - task: CmdLine@2
    displayName: Bundle msix files to msixbundle
    continueOnError: true
    inputs:
      script: |
        "C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64\makeappx.exe" bundle /d $(Build.ArtifactStagingDirectory)\ /p "$(Build.ArtifactStagingDirectory)\$(applicationName).msixbundle"

**Reply by Sergio0694** [reply]
> "it does mean the requirement to increase the TFM prevents distribution on lower windows versions via a standard installer or possibly the windows store"

No it does not.

**Reply by Joost-Jens-Luminis** [reply]
May I ask how you build an installer then? Maybe we are doing something wrong.

**Reply by Sergio0694** [reply]
I personally always use VS, not the command line. But regardless, have you checked the manifest in the generated package? Check the `MinVersion` attributes on the `TargetDeviceFamily` elements. Those are what should control this.

**Reply by Joost-Jens-Luminis** [reply]
Yes, and as expected for Windows.Desktop is indeed says 22621. We are not overriding this though in our pipeline so something else is. (The manifest we are providing in our solution say 19041)

**Reply by Joost-Jens-Luminis** [reply]
Looking through everything again, our Manifest file that we provide has the correct version number in it. We even make a copy of it for verification purposes (in our pipeline) so that is easy to see. I think the installer builder (MakeAppx) sets it wrong.

**Reply by Joost-Jens-Luminis** [reply]
I have found the solution. The Windows Project files requires
		&lt;TargetPlatformMinVersion&gt;10.0.19041.0&lt;/TargetPlatformMinVersion&gt;

The AppXManifest.xml file gets the MinVersion from this property. But with a WinUI app this property is not usually present in the .csproj file. Instead, it has the SupportedOSPlatformVersion property, however, this is not used for the AppXManifest.xml file and thus not used for the manifest of the Bundle.
Seems like a bug in .Net.
Thanks for your help on this.
